src/include/snapper/dialogs.rb

```diff
@@ -1233,6 +1018,196 @@ def ShowDialog
+                Opt(:hstretch),
+                _("Description"),
+                Ops.get_string(data, "description", "")
+              ),
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "userdata")),
+                Opt(:hstretch),
+                _("User data"),
+                Snapper.userdata_to_string(data["userdata"])
+              ),
+              Left(
+                ComboBox(
+                  Id(Ops.add(prefix, "cleanup")),
+                  Opt(:editable, :hstretch),
+                  # combo box label
+                  _("Cleanup algorithm"),
+                  cleanup_items(Ops.get_string(data, "cleanup", ""))
+                )
+              )
+            ),
+            HSpacing(0.4)
+          )
+        ),
+        HSpacing()
+      )
+    end
+
+    def update_snapshots
+      # busy popup message
+      Popup.ShowFeedback("", _("Reading list of snapshots..."))
+
+      Snapper.ReadSnapshots()
+
+      Popup.ClearFeedback
+
+      UI.ChangeWidget(Id(:snapshots_table), :Items, get_snapshot_items)
+      selected = UI.QueryWidget(Id(:snapshots_table), :SelectedItems) || []
+      enable_buttons([:modify, :show, :delete], selected.size == 1)
+      enable_buttons([:delete], selected.size >= 1)
+
+    end
+
+    def get_snapshot_items
+      snapshot_items = []
+
+      Snapper.snapshots.each_with_index do |s,i|
+        num = s["num"] || 0
+        start_date = (num != 0) ? timestring(s["date"]) : ""
+        end_date = ""
+        userdata = Snapper.userdata_to_string(s["userdata"])
+        desc = s["description"].to_s
+        if s["type"] == :SINGLE
+          type = _("Single")
+        elsif s["type"] == :POST
+          pre = s["pre_num"] || 0 # pre canot be 0
+          index = Ops.get(Snapper.id2index, pre, -1)
+          if pre == 0 || index == -1
+            Builtins.y2warning(
+              "something wrong - pre:%1, index:%2",
+              pre,
+              index
+            )
+            next
+          end
+          desc = Ops.get_string(Snapper.snapshots, [index, "description"], "")
+          end_date = start_date
+          start_date = timestring(Snapper.snapshots[index]["date"])
+          num = "%{pre} & %{post}" % { :pre => pre, :post => num }
+          type = _("Pre & Post")
+        else
+          # 0 means there's no post
+          if s["post_num"].to_i == 0
+            Builtins.y2milestone("pre snappshot %1 does not have post", num)
+            type = _("Pre")
+          else
+            Builtins.y2milestone("skipping pre snapshot: %1", num)
+            next
+          end
+        end
+        snapshot_items << Item(Id(i), num, type, start_date, end_date, desc, userdata)
+      end
+      snapshot_items
+    end
+
+    def config_select
+      HBox(
+        # combo box label
+        Label(_("Current Configuration")),
+        ComboBox(Id(:configs), Opt(:notify), "", Builtins.maplist(Snapper.configs) do |config|
```

Comment:
there it make sense to replace it with `map` and new line would be nice to use it. or add it above this expression like

```
items = Snapper.configs.map do |config|
  Item(Id(config), config, config == Snapper.current_config)
end
...
```